### ISSUE TITLE
upgrade docker openj9 to v0.11.0 to prevent JIT memory spikes triggering oom-killer

### ISSUE THREAD

**Issue by waxie** [linkerd/linkerd#2210]
## Filing a Linkerd issue ##

Issue Type:

- [x] Bug report
- [ ] Feature request

**What happened**:

When running linkerd 1.6.0 in docker (built with OpenJ9 0.8.0) image JIT compilations cause memory spikes beyond heap and container to go out-of-memory and get killed by kernel. This is from the buoyantio/linkerd docker image, tag 1.6.0-openj9-experimental:
```
OpenJDK Runtime Environment (build 1.8.0_162-b12)
Eclipse OpenJ9 VM (build openj9-0.8.0, JRE 1.8.0 Linux amd64-64 Compressed References 20180315_120 (JIT enabled, AOT enabled)
```

JIT compilations in OpenJ9 version 0.8.0 are not container/cgroup aware and take too much memory from the host memory, instead of the container memory. This issue is described here: 
https://blog.openj9.org/2018/06/12/eclipse-openj9-in-containers/

We would see sudden spikes of 1GB+ memory increase, causing the container to get oom-killed.

In https://www.eclipse.org/openj9/oj9_whatsnew.html it says for version 0.11.0:

> Memory optimizations for applications that run in containers We recently added some intelligence to make OpenJ9 "container-aware", which allows it to make informed decisions about the size of your Java heap if you run your application in a container. Since applications are typically run on their own in a container and do not need to compete for memory, the JVM allocates a larger proportion of memory for the Java heap. With more container memory available, the JIT compiler can work harder, improving the ramp up time for applications. Conversely, if we do not have as much container memory available, the JIT compiler can reduce its memory usage to fit within the container limits. Due to good performance results and positive feedback, we've now enabled this feature by default.

When I rebuild the docker image with OpenJ9 version 0.11.0 the memory spikes disappear and problem is resolved.

**What you expected to happen**:

Linkerd docker container to not get killed by oom-killer and not have memory spikes

**How to reproduce it (as minimally and precisely as possible)**:

Run linkerd 1.6.0 docker image tag `buoyantio/linkerd:1.6.0-openj9-experimental`

**Anything else we need to know?**:

I rebuilt the docker image now with: `adoptopenjdk/openjdk8-openj9:jdk8u192-b12_openj9-0.11.0` as Base.dockerJavaImage and that solved it:

`./sbt 'set Base.dockerJavaImage in (linkerd, OpenJ9) := "adoptopenjdk/openjdk8-openj9:jdk8u192-b12_openj9-0.11.0"' linkerd/openj9:docker
`
**Environment**:
- linkerd/namerd version, config files: `linkerd:1.6.0-openj9-experimental` `namerd-wrapped:1.6.0-openj9-experimental`
- Platform, version, and config files (Kubernetes, DC/OS, etc): Ubuntu 16.0.4.4, Docker-ce 17.06.2, Linux 4.15.0-33-generic
- Cloud provider or hardware configuration:

**Comment by dadjeibaah** [comment]
Thanks for filing this issue @waxie! We'll take a look and have this fixed.

**Comment by adleong** [comment]
Wow, great find, @waxie!  Are you interested in submitting a PR that updates our OpenJ9 base image?

**Comment by adleong** [comment]
Hey @waxie, I see that you opened and closed a PR for this.  Are you still interested in working on this or should we pick this up?

**Comment by dadjeibaah** [comment]
Fixed by #2232
